src/cljs/athens/views/blocks.cljs

```diff
@@ -221,7 +224,16 @@
 (defn on-change
   [value uid]
   ;; (prn "ONCHANGE" value)
-  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]]))
+  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]])
+  ;; automatically add non-existent pages
+  ;; TODO: delete pages that are no longer connected to anything else
+  (parse/transform {:page-link (fn [& title]
+                                 (if-not (db/search-exact-node-title (apply + title)) (let [uid (gen-block-uid)]
+                                                                                        (d/transact! db/dsdb [{:node/title     (str apply + title)
```

Comment:
Should use `(dispatch [:transact])` pattern like above.

Reply:
Fixed!